TechnicalDocuments/Proposals/BabylonDesignLibrary.md

```diff
@@ -0,0 +1,136 @@
+# Babylon Design Library
+
+* Author(s): Giorgos Tsiapaliokas
+* Review Manager: Anders Ha
+
+## Introduction
+This proposal is about introducing a Design Library.
+
+## Motivation
+With our current infrastrure we have have reusable components which don't reuse the same style,
+we provide different styesheets based on the design.
+
+The idea behind this is to have reusable components with a predefined stylesheet displaying different data.
+The benefit of having a design library is
+1. to develop new UIs faster.
+2. to catch inconsistencies in the UI due to a deisgn flaw or due to an engineering mistake.
+
+The requirement is to cover most of the common use cases and *not* to extend to edge cases.
+Also we must reuse our current tools like `Bento`.
+
+For instance, in our codebase a common UI element is a rounded button with background color. 
+For that we use `Component.Button` and a stylesheet while this approach is functional is has a few drawbacks.
+The attributes of the button (background color, alpha, corner radius etc) can either 
+1. have been redefined in another part of the codebase which is good from a design point of view *and if* we search
+for them we can use them, *if we don't* search for them we will just reimplement something which already exists.
+2. or the attributes of the button haven't been redefined in another part of the application and we may not
+notice that so a design flaw will be implemented. We are humans after all :).
+
+
+## Proposed solution
+We can introduce `AppearenceService` which will be responsible for creating these components following
```

Comment:
[nit] It's not a "service" in the meaning we put in other services. Just calling it `Appearance` would be enough.

Reply:
I don't have strong feelings about the name.
I just picked something.

Just `Appearance ` isn't ok since its `AppearanceService. Appearance`

Reply:
`AppearanceContainer` ? 
Since it will responsible for providing `AppearanceContainer. Appearance`.

Reply:
Why do we even need such level of nesting? Why `typography` or `colours` can't be a property of `Appearance` which would be a top level type?
If we want to keep colors & fonts and separate from Bento components then there should be `Appearance` for just fonts and colors and components factory that would be created with a an instance of appearance.